lib/internal/modules/cjs/loader.js

```diff
@@ -84,6 +84,20 @@ const relativeResolveCache = Object.create(null);
 
 let requireDepth = 0;
 let statCache = null;
+
+function enrichCJSError(err) {
+  const stack = err.stack.split('\n');
+
+  const lineWithErr = stack[1];
+
+  if (err.message.startsWith('Unexpected token export') ||
+    (/^\s*import(?=[ {'"*])\s*(?![ (])/).test(lineWithErr)) {
```

Comment:
nit: this wouldn't handle minified, line split imports, etc. would it? e.g.

```mjs
import
  'fs';
```

```mjs
var x; import 'fs';
```

```mjs
/**/import 'fs';
```


could we just add a comment about this being "best effort for the common usage" and not comprehensive; comprehensive parsing is probably overkill for the cost/gains, but when people come back to this code, it would give context. I don't want to come back to this and have it stated this is a guaranteed kind of message that will be produced unless we do want to do comprehensive parsing.

Reply:
Correct, cases like those you mention are not matched by that regexp introduced in this pull request, indeed targeting such cases might cause performance penalty or even errors by attempting to match something in the string.

In the beginning, we tried to cover as many as possible but often we encountered edge cases and the regexp became overcomplicated.

That's why we decided to cover as many cases as possible and leave such cases not covered.

A comment like 

```
  /*
    The regular expression below targets the most common import statement usage, however, there are cases that are not matching, cases like import statement after a comment block and/or after a variable definition.
  */
```
above the condition would make it more understandable, right?

Reply:
Sounds good